src/simplewallet/simplewallet.cpp

```diff
@@ -602,12 +616,14 @@ std::string simple_wallet::get_command_usage(const std::vector<std::string> &arg
 
 bool simple_wallet::viewkey(const std::vector<std::string> &args/* = std::vector<std::string>()*/)
 {
-  if (m_wallet->ask_password() && !get_and_verify_password()) { return true; }
+  SCOPED_WALLET_UNLOCK();
   // don't log
   if (m_wallet->key_on_device()) {
     std::cout << "secret: On device. Not available" << std::endl;
   } else {
-    std::cout << "secret: " << string_tools::pod_to_hex(m_wallet->get_account().get_keys().m_view_secret_key) << std::endl;
+    printf("secret: ");
+    print_secret_key(m_wallet->get_account().get_keys().m_view_secret_key);
+    putchar('\n');
```

Comment:
When `ask-password` is disabled, the output text becomes slightly strange:
```
[wallet 55AG7C]: viewkey
Wallet password: 
secret: d042f7012cafc1aecf89b06daed52f9fdbc17ea75e8d03aec36613b4fccb0102
public: e7fb43ed085528d3efce5e2ce054b428bbedcb80b2ed86ef1c090fbeb9fa013b
[wallet 55AG7C]: set ask-password 0
Wallet password: 
[wallet 55AG7C]: viewkey
[wallet 55AG7C]: secret: d042f7012cafc1aecf89b06daed52f9fdbc17ea75e8d03aec36613b4fccb0102
public: e7fb43ed085528d3efce5e2ce054b428bbedcb80b2ed86ef1c090fbeb9fa013b
```
I've tried a few things but couldn't figure out how to fix it.